cloudnet-modules/cloudnet-database-mongodb/src/main/java/de/dytanic/cloudnet/ext/database/mongodb/MongoDatabaseProvider.java

```diff
@@ -0,0 +1,149 @@
+    }
+
+    @Override
+    public boolean init() throws Exception {
+        Executors.newScheduledThreadPool(1).scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                if (!isConnected()) {
+                    try {
+                        closeSession();
+                    } catch (Exception ex) {
+                    }
+                    connect();
+                }
+            }
+
+        }, 0, 3, TimeUnit.MINUTES);
+        return true;
+    }
+
+    public void connect() {
+        try {
+            this.client = MongoClients.create(new ConnectionString(this.databaseConnectionData.toConnectionString()));
+        } catch (Exception ex) {
+            CloudNetDriver.getInstance().getLogger().error(ex.getMessage(), ex);
+        }
+    }
+
+    @Override
+    public String getName() {
+        return "MongoDB-Provider";
+    }
+
+    public MongoDatabase getMongoDatabase() {
+        if (getClient() == null) {
+            connect();
+        }
+        return getClient().getDatabase(getDatabaseConnectionData().getDatabase());
+    }
+
+    @Override
+    public Database getDatabase(String name) {
+        return new de.dytanic.cloudnet.ext.database.mongodb.MongoDatabase(this, name, executorService);
+    }
+
+    @Override
+    public boolean containsDatabase(String name) {
+        return client.getDatabase(name) != null;
+    }
+
+    @Override
+    public @NotNull ITask<Boolean> containsDatabaseAsync(String name) {
+        return this.schedule(() -> containsDatabase(name));
+    }
+
+    @Override
+    public @NotNull ITask<Boolean> deleteDatabaseAsync(String name) {
+        return this.schedule(() -> deleteDatabase(name));
+    }
+
+    @Override
+    public @NotNull ITask<Collection<String>> getDatabaseNamesAsync() {
+        return this.schedule(() -> getDatabaseNames());
+    }
+
+    private <T> ITask<T> schedule(Callable<T> callable) {
+        ITask<T> task = new ListenableTask<>(callable);
+        this.executorService.execute(() -> {
+            try {
+                Thread.sleep(0, 100000);
+                task.call();
+            } catch (Exception ex) {
+                CloudNetDriver.getInstance().getLogger().error(ex.getMessage(), ex);
+            }
+        });
+        return task;
+    }
+
+    public boolean deleteDatabase(String name) {
+        if (client.getDatabase(name) != null) {
+            client.getDatabase(name).drop();
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public Collection<String> getDatabaseNames() {
+        Collection<String> names = new ArrayList<>();
+        client.getDatabase(this.getDatabaseConnectionData().getDatabase()).listCollectionNames().forEach(n -> {
+            names.add(n);
+        });
+        return names;
+    }
+
+    @Override
+    public void close() throws Exception {
+        if (client != null) {
+            client.close();
+        }
```

Comment:
What is the diffrence to `closeSession()` or why is this not calling `closeSession()`?